File: c/c99/1061_tempo_de_um_evento.c
#include <stdio.h>

int main() {
    int diaInicio, h1, m1, s1;
    int diaFim, h2, m2, s2;

    scanf("Dia %d", &diaInicio);
    scanf("%d : %d : %d", &h1, &m1, &s1);
    scanf("Dia %d", &diaFim);
    scanf("%d : %d : %d", &h2, &m2, &s2);

    int inicioSeg = s1 + m1 * 60 + h1 * 3600 + diaInicio * 86400;
    int fimSeg = s2 + m2 * 60 + h2 * 3600 + diaFim * 86400;

    int duracao = fimSeg - inicioSeg;

    int dias = duracao / 86400;
    duracao %= 86400;
    int horas = duracao / 3600;
    duracao %= 3600;
    int minutos = duracao / 60;
    int segundos = duracao % 60;

    printf("%d dia(s)\n", dias);
    printf("%d hora(s)\n", horas);
    printf("%d minuto(s)\n", minutos);
    printf("%d segundo(s)\n", segundos);

    return 0;
}

File: python/1 - Iniciante/1061_tempo_de_um_evento.py
inicio = int(input().split()[1])
h1, m1, s1 = map(int, input().split(' : '))
fim = int(input().split()[1])
h2, m2, s2 = map(int, input().split(' : '))

dias = fim - inicio

if s1 <= s2:
    segundos = s2 - s1

else:
    segundos = (60 - s1) + s2
    m2 -= 1

if m1 <= m2:
    minutos = m2 - m1

else:
    minutos = (60 - m1) + m2
    h2 -= 1

if h1 < h2:
    horas = h2 - h1

elif h1 == h2 and m1 == m2 and s1 == s2:
    horas = 0
    minutos = 0
    segundos = 0
    dias += 1

else:
    horas = (24 - h1) + h2
    dias -= 1

if horas == 24:
    horas = 0
    dias += 1

print(f'{dias} dia(s)')
print(f'{horas} hora(s)')
print(f'{minutos} minuto(s)')
print(f'{segundos} segundo(s)')
